lumeer-core/src/main/java/io/lumeer/core/PermissionsChecker.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * -----------------------------------------------------------------------\
+ * Lumeer
+ *  
+ * Copyright (C) since 2016 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.core;
+
+import io.lumeer.core.exception.NoPermissionException;
+import io.lumeer.api.model.Permission;
+import io.lumeer.api.model.Resource;
+import io.lumeer.api.model.Role;
+import io.lumeer.storage.api.dao.UserDao;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+@RequestScoped
+public class PermissionsChecker {
+
+   @Inject
+   private UserDao userDao;
+
+   @Inject
+   private AuthenticatedUser authenticatedUser;
+
+   public PermissionsChecker() {
+   }
+
+   PermissionsChecker(UserDao userDao, AuthenticatedUser authenticatedUser) {
+      this.userDao = userDao;
+      this.authenticatedUser = authenticatedUser;
+   }
+
+   /**
+    * Checks if the user has the given role on the given resource (either directly or through group membership).
+    *
+    * @param resource
+    *       any resource with defined permissions
+    * @param role
+    *       role to be checked
+    * @throws NoPermissionException
+    */
+   public void checkRole(Resource resource, Role role) {
+      if (!hasRole(resource, role)) {
+         throw new NoPermissionException(resource);
+      }
+   }
+
+   public boolean hasRole(Resource resource, Role role) {
+      return getActualRoles(resource).contains(role);
+   }
+
+   private boolean hasUserRole(Set<Permission> userRoles, String user, Role role) {
+      return userRoles.stream()
+                      .filter(entity -> entity.getName().equals(user))
+                      .flatMap(entity -> entity.getRoles().stream())
+                      .anyMatch(actualRole -> actualRole.equals(role));
+   }
```

Comment:
Oh, these are the old methods that are no longer needed since they alternatives has already been implemented using `getActualRoles`.